backend/stan-wasm-server/src/app/logic/compilation.py

```diff
@@ -18,10 +19,27 @@
 logger = logging.getLogger(__name__)
 
 
+@lru_cache
+def _get_salt() -> bytes:
+    """
+    Returns a salt to use when hashing Stan programs if one was set during the Docker build.
+    This helps ensure that models don't get cached by the browser across different deployments.
+    """
+    salt_file = Path(__file__).parent.parent.parent.parent / "hash-salt.txt"
+    if not salt_file.exists():
+        logger.warning("No hash salt file found at %s", salt_file)
+        salt = b""
+    else:
+        salt = salt_file.read_bytes().strip()
+        logger.info("Using hash salt from %s: %s", salt_file, repr(salt))
+    return salt
+
```

Comment:
Should I be seeing the side effect of this function every time I compile a new (cache miss) `main.stan`? I would've thought these would be cache hits, but maybe I need a refresher on our threading model here.

Reply:
Actually, it's not even that predictable when the function reruns and when it doesn't.

I don't think this much matters, but I like to note everything.

Reply:
Is there any rhyme or reason to when a new thread is spawned?

Reply:
Not sure, that would be a uvicorn implementation detail. It seems like it’s roughly round-robin